### PR TITLE
docs: clarify MLLM MTP guard semantics

### LINKED ISSUE #489: Follow up PR #397 MTP guard and stream-binding review notes
## Context

PR #397 fixed important SpecPrefill/MTP and MLX stream-thread ownership issues, but it was self-merged after comment-only review. The review identified architecture/performance concerns that should be tracked explicitly.

## Review notes to resolve

- The batched MLLM MTP path currently disables MTP for non-greedy sampling (`temperature != 0` or `top_p < 1`). That is correctness-safe, but may make MTP effectively greedy-only for common production sampling settings. This needs either a narrower guard, evidence-backed performance decision, or clear documentation as a known limitation.
- Verify/document the processor-retirement continuation contract and duplicated helper state after later refactors.
- Confirm `_bind_worker_generation_streams()` per-call stream creation does not cause memory growth under repeated requests, or document the behavior.
- Confirm `prefill_step_size` fallback behavior for MLLM is intentional and documented.

## Work to do

- Add focused tests or docs for the non-greedy MTP guard semantics.
- Add a small stress/telemetry check or explanation for worker stream binding behavior.
- Close as clean only with explicit evidence.

## Scope note

This is not a revert request and not a broad MTP failure claim. It is review debt from #397 that should not remain only in PR comments.

### PR DESCRIPTION
**Pull request by Thump604**
Closes #489.

This is a narrow follow-up for the PR #397 review notes. I am not changing MLLM MTP behavior in this PR; I am making the current safe envelope explicit and tightening regression coverage around the existing guard.

Changes:
- document that batched MLLM MTP is currently a conservative greedy-only optimization
- document that request-local processors stay active by default and retirement-to-MTP resume remains explicit opt-in via `VLLM_MLX_ENABLE_THINKING_RETIREMENT_RESUME=1`
- document MLLM `prefill_step_size` fallback behavior and worker stream rebinding scope
- expand the MLLM MTP guard test so `temperature`, `top_p`, `top_k`, and `min_p` each independently fall back to the normal scheduler path
- clarify `bind_generation_streams()` as a worker-entry thread-ownership guard, not a token-loop optimization

Validation:
- `AI_RUNTIME_BYPASS_SAFETY_GATE=1 /opt/ai-runtime/venv-live/bin/python -m pytest tests/test_mllm_continuous_batching.py -q -k 'MTPGuards or BatchedMLLMConfigWiring'`
- `AI_RUNTIME_BYPASS_SAFETY_GATE=1 /opt/ai-runtime/venv-live/bin/python -m pytest tests/test_simple_engine.py tests/test_simple_engine_cancel_serialization.py tests/test_engine_core_thread_streams.py -q -k 'stream or serialized or bind_generation_streams or prelock'`
- `AI_RUNTIME_BYPASS_SAFETY_GATE=1 /opt/ai-runtime/venv-live/bin/python -m compileall -q vllm_mlx/mllm_batch_generator.py vllm_mlx/mlx_streams.py`
- `black --target-version py312 --check vllm_mlx/mlx_streams.py tests/test_mllm_continuous_batching.py`
- `uvx ruff check vllm_mlx/mlx_streams.py tests/test_mllm_continuous_batching.py --select E,F,W --ignore E402,E501,E731,F811,F841`
- `git diff --check`

### CODE REVIEW

**Review by janhilgard — APPROVED** [review]
Clean docs-and-tests-only PR. No behavioral changes, just documentation and better regression coverage.

**Docs** — the continuous-batching guide section accurately describes the current MLLM MTP envelope (greedy-only, single-request batch, no logits processors), the opt-in retirement-to-MTP handoff, prefill step size semantics, and stream rebinding scope. The `bind_generation_streams()` docstring clarification ("ownership fix, not a batching optimization; call at worker-entry boundaries") is valuable given the ongoing #478/#479 worker-thread discussion.

**Tests** — parametrizing the non-greedy test across all four sampling dimensions (temperature, top_p, top_k, min_p individually) is a solid improvement over the old test that set all four simultaneously. This ensures each guard is independently verified — a regression that only breaks the `top_k` check would have been invisible before.

Single commit, CI green. LGTM.

**Comment by waybarrios** [comment]
Took a local look. Docs line up with the actual guard in `mllm_batch_generator.py` and the env var checks in `engine/simple.py`. Test parametrization is a nice catch, way better than the old all-at-once version.

Merging. Thanks!
